### ISSUE TITLE
Unrecognized function or variable 'lightsWritten'

### ISSUE THREAD

**Issue by AmyMNi** [ISET/iset3d#46]
Hi all, 

When running t_piIntro_light.m on the dev branch, I get the following error:

Unrecognized function or variable 'lightsWritten'.

Error in piWrite>piIncludeLines (line 664)
        if piContains(currLine, 'WorldEnd')  && lightsWritten == false

Error in piWrite (line 193)
piIncludeLines(thisR,fileID, creatematerials,overwritegeometry);

Error in t_piIntro_light (line 92)
piWrite(thisR);

I first got this error when trying to render my natural scenes, which I how I came across it. Thanks in advance!
Amy

**Comment by wandell** [comment]
I think master is running at this point.  At least v_iset3d runs fo4 me.

Also, can you give the sceneauto branch a try, please.
If we can get you running on Blender and everything else on that branch, life will be much simpler.

I see the error on dev.  I will try to figure it out tomorrow.

--------------------
Sent from my iPad
________________________________
From: Amy Ni ***@***.***>
Sent: Wednesday, June 2, 2021 10:26:00 PM
To: ISET/iset3d ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [ISET/iset3d] Unrecognized function or variable 'lightsWritten' (#46)


Hi all,

When running t_piIntro_light.m on the dev branch, I get the following error:

Unrecognized function or variable 'lightsWritten'.

Error in piWrite>piIncludeLines (line 664)
if piContains(currLine, 'WorldEnd') && lightsWritten == false

Error in piWrite (line 193)
piIncludeLines(thisR,fileID, creatematerials,overwritegeometry);

Error in t_piIntro_light (line 92)
piWrite(thisR);

I first got this error when trying to render my natural scenes, which I how I came across it. Thanks in advance!
Amy

**Comment by AmyMNi** [comment]
Thank you, testing sceneauto branch now with Blender export. I'm not able to add a light (which is to say, no error when using piLightCreate, but the rendered scene is an all black image). Will try to track this down.

**Comment by AmyMNi** [comment]
Update: I'm able to add an infinite light and render an all white image, so the problem is not with adding lights.

**Comment by wandell** [comment]
@ZhengLyuFelix is the one who solves my problems when we get to this stage.  But I am glad to Zoom with you to try to track down the issue.  Does the master branch work, by the way?

**Comment by AmyMNi** [comment]
Yes, master branch still works. I’m happy to work on the sceneauto branch
to get it working with Blender exports, though it won’t be today.

Best,
Amy

**Comment by wandell** [comment]
We tagged the master version that runs with Blender and saved it, so we will always have that.

I will try to work with Zheng on the Blender script to see if we can get it working without your intervention on sceneauto.  He is headed off to Apple soon, but maybe we can get it done before that!

**Comment by ZhengLyuFelix** [comment]
Thanks for the notification, yes we the logic of that section in dev branch is not robust. We think (hope) that the problem has been fixed in sceneauto. WIll work with Brian on it.

**Comment by AmyMNi** [comment]
Thank you! Congrats to Zheng! I’ll also stay in touch if I find anything
useful.

Best,
Amy
